snuba/migrations/migrations.py

```diff
@@ -0,0 +1,91 @@
+import logging
+
+from datetime import datetime
+from enum import Enum
+from importlib import import_module
+from typing import Optional, Sequence, Tuple
+
+from snuba.clusters.cluster import ClickhouseClientSettings, get_cluster, CLUSTERS
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.migrations.operations import Operation
+
+
+logger = logging.getLogger("snuba.migrations")
+
+LOCAL_TABLE_NAME = "migrations_local"
+DIST_TABLE_NAME = "migrations_dist"
+
+# Currently only local migrations are supported
+TABLE_NAME = LOCAL_TABLE_NAME
+
+
+class App(Enum):
+    SYSTEM = "system"
+    SNUBA = "snuba"
+
+
+class Status(Enum):
+    NOT_STARTED = "not_started"
+    IN_PROGRESS = "in_progress"
+    COMPLETED = "completed"
+
+
+class Migration:
```

Comment:
Should we partition migrations per dataset ? Dataset are independent partitions of the datamodel so they could own their own migrations. I think this would logically make sense, the question is whether migrations for a given dataset should be able to run inndependently from each other dataset, if there is a value in that beyond data partitioning and how to sequence them properly with respect to system migrations.

Reply:
I'm not sure exactly what you mean by "partition" the migration here. Are you talking about partitioning the ClickHouse table or something else? Also doesn't this depend on a major overhaul of our datasets that is still fairly uncertain since discover, events, errors, transactions for example are currently all defined as different datasets.